### PR TITLE
aws - timestream-table, timestream-database - add resources

### LINKED ISSUE #8156: Support for AWS Timestream
### Describe the feature

Support for AWS timestream that includes adding support for tagging and deletion of resource

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by thisisshi**
closes #8156

### CODE REVIEW

File: c7n/resources/timestream.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        enum_spec = ('list_tables', 'Tables', {})
+        permission_prefix = 'timestream'
+        permissions = ('timestream:ListTables', )
+
+    source_mapping = {
+        'describe': DescribeTimestream,
+    }
+
+
+@TimestreamDatabase.action_registry.register('tag')
```
Comment (PratMis) [inline c7n/resources/timestream.py:53]: Looks like `timestream-database` is supported in resourcegroupstaggingapi. We could potentially use that to read/write tags.
```
>>> response = client.tag_resources(ResourceARNList=['arn:aws:timestream:us-east-1:12356789123:database/sampleDB'], Tags={'c7n': 'test'})
>>> response = client.get_resources(ResourceARNList=['arn:aws:timestream:us-east-1:12356789123:database/sampleDB'])
>>> response
{'PaginationToken': '', 'ResourceTagMappingList': [{'ResourceARN': 'arn:aws:timestream:us-east-1:471176887411:database/sampleDB', 'Tags': [{'Key': 'Owner', 'Value': 'xyz'}, {'Key': 'c7n', 'Value': 'test'}]}]}
```

File: c7n/resources/timestream.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+from c7n.utils import local_session, type_schema
+from c7n.tags import (
+    TagDelayedAction,
+    TagActionFilter,
+    Tag as TagAction,
+    RemoveTag as RemoveTagAction
+)
+
+
+class DescribeTimestream(DescribeSource):
```
Comment (PratMis) [inline c7n/resources/timestream.py:13]: We could use the RGT api here to read tags for both `timestream-database` and `timestream-table`.

File: c7n/resources/timestream.py (hunk 1 above)
Comment (thisisshi) [reply c7n/resources/timestream.py:53]: weirdly enough i'm unable to tag via the resourcegroupstagging api in my account:

```
{'FailedResourcesMap': {'arn:aws:timestream:us-east-1:123456789012:database/test': {'StatusCode': 400, 'ErrorCode': 'InvalidParameterException', 'ErrorMessage': 'Unrecognized service or resource type for tagging'}}, 'ResponseMetadata': {'RequestId': '41827cc7-c48a-482e-9550-5cab0f87f569', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': '41827cc7-c48a-482e-9550-5cab0f87f569', 'content-type': 'application/x-amz-json-1.1', 'content-length': '208', 'date': 'Wed, 25 Jan 2023 22:34:16 GMT'}, 'RetryAttempts': 0}}
```

File: c7n/resources/timestream.py (hunk 2 above)
Comment (thisisshi) [reply c7n/resources/timestream.py:13]: see my comment above, basically i'm not convinced that timestream is _fully_ supported with resourcegroupstaggingapi i would note that it also doesnt show up in my tag explorer in the aws console

File: c7n/resources/timestream.py (hunk 1 above)
Comment (PratMis) [reply c7n/resources/timestream.py:53]: you're right, I tested in some external accounts and it didn't work

**Review by PratMis — APPROVED** [review]
Thanks @thisisshi , lgtm

**Pull request merged**